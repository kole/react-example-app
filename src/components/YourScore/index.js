import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const YourScoreContainer = styled.div`
    font-size: 3em;
    font-weight: bold;
`;

const YourScore = (props) => {
    const { correctAnswerCount, totalQuestions } = props;
    return (
        <YourScoreContainer>
            You scored <br />{correctAnswerCount} / {totalQuestions}
        </YourScoreContainer>
    );
}

YourScore.propTypes = {
    correctAnswerCount: PropTypes.number.isRequired,
    totalQuestions: PropTypes.number.isRequired
}

export default YourScore;
