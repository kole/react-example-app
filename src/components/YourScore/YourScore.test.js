/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';
import YourScore from './index.js';

describe('YourScore', () => {
    test('Snapshot', () => {
        const component = renderer
            .create(<YourScore correctAnswerCount={5} totalQuestions={10} />)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
});
