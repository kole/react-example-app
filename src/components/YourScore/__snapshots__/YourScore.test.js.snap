// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YourScore Snapshot 1`] = `
<div
  className="sc-bdVaJa krsHcM"
>
  You scored 
  <br />
  5
   / 
  10
</div>
`;
